connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1275,6 +1276,24 @@ class Dataset[T] private[sql] (
       proto.Aggregate.GroupType.GROUP_TYPE_GROUPBY)
   }
 
+  /**
+   * (Scala-specific)
+   * Returns a [[KeyValueGroupedDataset]] where the data is grouped by the given key `func`.
+   *
+   * @group typedrel
+   * @since 2.0.0
+   */
+  def groupByKey[K: Encoder](func: T => K): KeyValueGroupedDataset[K, T] = {
+
+    new KeyValueGroupedDataset(
+      encoderFor[K],
+      encoderFor[T],
+      sparkSession,
+      plan,
+      Seq.empty,
```

Comment:
Set to empty for both `dataAttributes` and `groupingAttributes` since I didn't find a good way to get them. From the existing implementation [here](https://github.com/apache/spark/blob/16a28b1a961052a250dcf05b7c249c92156e1077/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L1939-L1947), they are retrieved from `logicalPlan`. But since Spark Connect's `Dataset` is not able to get the `logicalPlan`, should we remove `dataAttributes` and `groupingAttributes` from the constructor and move the logic to `transformFlatMapGroupsWithState` in `SparkConnectPlanner`?